validator/appengine_location.rego

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPAppengineLocationConstraintV1
+
+import data.validator.gcp.lib as lib
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	constraint := input.constraint
+	lib.get_constraint_params(constraint, params)
+
+	asset := input.asset
+
+	# Applies to appengine applications only
+	asset.asset_type == "appengine.googleapis.com/Application"
+
+	# Retrieve the list of allowed locations - default to empty set
+	locations := cast_set(lib.get_default(params, "locations", {}))
```

Comment:
If locations are a required param, we shouldn't set a default.

Reply:
updated too. Thanks!